injector/src/main/java/edu/ucr/cs/riple/injector/changes/AddSingleElementAnnotation.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2022 University of California, Riverside.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.changes;
+
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.ArrayInitializerExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.Name;
+import com.github.javaparser.ast.expr.SingleMemberAnnotationExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.github.javaparser.ast.nodeTypes.NodeWithAnnotations;
+import com.google.common.base.Preconditions;
+import edu.ucr.cs.riple.injector.location.Location;
+import java.util.Optional;
+
+/**
+ * Used to add <a
+ * href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-SingleElementAnnotation">Single
+ * Element Annotations</a> with only one member on elements in source code. If the annotation
+ * already exists, it can be collapsed into a single annotation with multiple elements if requested.
+ */
+public class AddSingleElementAnnotation extends AddAnnotation {
+  /** Argument of the annotation. */
+  private final String argument;
+
+  /**
+   * If true, if an annotation with the same name exists, the existing annotation should include the
+   * passed argument, otherwise, a new annotation will be injected to the node.
+   */
+  private final boolean collapseArguments;
+
+  public AddSingleElementAnnotation(
+      Location location, String annotation, String argument, boolean collapseArguments) {
+    super(location, annotation);
+    Preconditions.checkArgument(
+        argument != null && !argument.equals(""),
+        "argument cannot be null or empty, use AddAnnotation instead.");
```

Comment:
I understand not allowing null (Is this code being checked by NullAway, though? Because if so, we don't need the runtime check 😉 ). But, why is empty string not allowed? `@Foo("")` is a valid annotation if `Foo` is a Single-Element Annotation.

Reply:
Not now, but It will be checked by `NullAway` as a long-term plan mentioned in #21 in future. I think it does not make sense to add the annotation as `@Foo("")`. Maybe we can force it to use `MarkerAnnotation` in such cases ? However, if you think still it might be useful in some scenarios to support such injections, please let me know and I will remove the guard. Thank you.

Reply:
Let's leave it like this for now. I don't think there is anything in the Java spec that prevents you from having a single element annotation with the empty string as its `value` field, and there is definitely a semantic difference between empty-string vs no-value, but I don't think we care for any of the foreseeable use cases of the auto-annotator, so let's leave the check in for now.

Reply:
Ok great sounds great. Will update it in future if needed.